Depcrecate Configuration.isVisible() (#34079)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

=== Deprecations

> 

Reviewer: Extra line?

}
----

[[deprecate-visible-property]]
==== The `Configuration.visible` property is now deprecated

> Prior to Gradle 9.0, the artifacts of any configuration where `isVisible()` returned true would be built whenever the `assemble` task was executed.

Reviewer 2: ```suggestion
Prior to Gradle 9.0, any configuration with `isVisible()` returning `true` would have its artifacts built automatically when the `assemble` task was executed.
```

> This implicit behavior was removed in 9.0, and the `Configuration.visible` property no longer has any meaningful effect.

Reviewer 2: ```suggestion
This implicit behavior was removed in 9.0.0, and the `Configuration.visible` property no longer has any meaningful effect.
```

This property is now deprecated and will be removed in Gradle 10.0.0.

> Any usage of this property can be removed.

Reviewer 2: ```suggestion
Any usage of `visible` can be removed.
```

> If the artifacts of the configuration should be built by the `assemble` task, create an explicit task dependency on the `assemble` task instead.

Reviewer 2: ```suggestion
If you want the artifacts of a configuration to be built as part of the `assemble` task, add an explicit task dependency on `assemble` instead:
```
